src/devices/Mcp3428/Helpers.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Mcp3428
+{
+    public enum GainEnum : byte { X1 = 0, X2 = 1, X4 = 2, X8 = 3 }
+
+    public enum ModeEnum : byte { OneShot = 0, Continuous = 16 }
+
+    public enum PinState : byte { Low = 0, High = 1, Floating = 2 }
+
+    public enum ResolutionEnum : byte { Bit12 = 0, Bit14 = 4, Bit16 = 8 }
+
+    internal static class Helpers
+    {
+        public static double LSBValue(ResolutionEnum res)
+        {
+            switch (res)
+            {
+                case ResolutionEnum.Bit12:
+                    return 1e-3;
+
+                case ResolutionEnum.Bit14:
+                    return 250e-6;
+
+                case ResolutionEnum.Bit16:
+                    return 62.5e-6;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(res), res, null);
+            }
+        }
+
+        // From datasheet 5.2
+        /// <summary>
+        /// Address from pin configuration. Based on documentation TABLE 5-3-
+        /// </summary>
+        /// <param name="Adr1">The adr1.</param>
+        /// <param name="Adr0">The adr0.</param>
+        /// <returns>System.Byte.</returns>
+        /// <exception cref="ArgumentException">Invalid combination</exception>
+        /// <autogeneratedoc />
+        public static byte AddressFromPins(PinState Adr0, PinState Adr1)
```

Comment:
I think the name should be clearer what this address refers to

Reply:
Renamed to I2CAddressFromPins + better doc